rbx_binary/src/serializer/state.rs

```diff
@@ -449,7 +449,7 @@ impl<'dom, 'db, W: Write> SerializerState<'dom, 'db, W> {
                 // There's no assurance that the default SharedString value
                 // will actually get serialized inside of the SSTR chunk, so we
                 // check here just to make sure.
-                if let Cow::Owned(Variant::SharedString(sstr)) = &default_value {
+                if let Variant::SharedString(sstr) = default_value.borrow() {
```

Comment:
Turns out we were matching against `Cow::Owned` when checking if we need to insert default SharedStrings instead of just borrowing